src/components/ModalEditCourse/index.tsx

```diff
@@ -0,0 +1,164 @@
+    course: {
+      id: courseEditMeta?.id || '',
+      name: courseName,
+      teamSize: +teamSize,
+      isActive: isCourseActive,
+    },
+  });
+
+  const onCloseModal = useCallback(() => {
+    setCourseEditMeta(null);
+    setCourseNameErrorMessage('');
+    setTeamSizeErrorMessage('');
+    dispatch(activeModalEditCourse(false));
+  }, [dispatch, setCourseEditMeta]);
+
+  const onSubmitModal = useCallback(() => {
+    if (isCourseNameFieldValid && isTeamSizeFieldValid) {
+      if (courseName && teamSize) {
+        const isCourseChanged = courseEditMeta
+          ? isCourseInfoChanged(courseEditMeta, courseName, teamSize, isCourseActive)
+          : true;
+        isCourseChanged && updateCourse();
+        onCloseModal();
+      } else {
+        if (!courseName) {
+          setIsCourseNameFieldValid(false);
+          setCourseNameErrorMessage(COURSE_NAME_VALIDATION.minLength.message);
+        }
+        if (!teamSize) {
+          setIsTeamSizeFieldValid(false);
+          setTeamSizeErrorMessage(TEAM_SIZE_VALIDATION.minLength.message);
+        }
+      }
+    }
+  }, [
+    onCloseModal,
+    courseEditMeta,
+    courseName,
+    isCourseActive,
+    isCourseNameFieldValid,
+    isTeamSizeFieldValid,
+    teamSize,
+    updateCourse,
+  ]);
+
+  useEffect(() => {
+    const listener = (e: KeyboardEvent) => {
+      if (e.key === 'Enter') {
+        onSubmitModal();
+      }
+    };
+    if (open) {
+      document.addEventListener('keydown', listener);
+    }
+    return () => document.removeEventListener('keydown', listener);
+  }, [onSubmitModal, open]);
+
+  const changeCourseStateButtonText = isCourseActive ? 'Terminate course' : 'Activate course';
+  const onChangeCourseState = () => setIsCourseActive(!isCourseActive);
+
+  if (errorUpdateCourse) return <ErrorModal />;
+
+  return (
+    <Modal
+      {...{ title, text, open, okText, cancelText }}
+      onClose={onCloseModal}
+      onSubmit={onSubmitModal}
+      hideOnOutsideClick
+      hideOnEsc
+    >
+      <InputBlock
+        inputLabel="New course name"
+        value={courseName}
+        placeholder="Enter new course name"
+        onChangeHandler={onChangeField(
+          COURSE_NAME_VALIDATION,
+          setIsCourseNameFieldValid,
+          setCourseNameErrorMessage,
+          setCourseName,
+          isCourseNameUniq
+        )}
+        isFieldValid={isCourseNameFieldValid}
+        errorMessage={courseNameErrorMessage}
+        isModal
+      />
+      <InputBlock
+        inputLabel="New team size"
+        value={teamSize}
+        placeholder="Enter new team size"
+        onChangeHandler={onChangeField(
+          TEAM_SIZE_VALIDATION,
+          setIsTeamSizeFieldValid,
+          setTeamSizeErrorMessage,
+          setTeamSize
+        )}
+        isFieldValid={isTeamSizeFieldValid}
+        errorMessage={teamSizeErrorMessage}
+        isModal
+      />
+      <TeamButton
```

Comment:
i think it should be someway highlighted with css as a button. looks a little weird as for me now.

Reply:
fixed